### ISSUE TITLE
'make check fails to link: error: only libraries may -dlpreopen a convenience library: './modules/python/libmod-python.la'

### ISSUE THREAD

**Issue by yurivict** [syslog-ng/syslog-ng#4230]
# syslog-ng
## Version of syslog-ng
3.38.1

## Platform
FreeBSD 13.1

## Debug bundle

# Issue
## Failure
```
/bin/sh ./libtool  --tag=CC   --mode=link cc -DTOP_SRCDIR=\"/disk-samsung/freebsd-ports/sysutils/syslog-ng/work/syslog-ng-3.38.1\" -Wshadow -Wno-stack-protector -Wno-unused-parameter -Wno-variadic-macros   -I/usr/local/include -DBXF_STATIC_LIB=1   -I./modules/python -O2 -g -pthread -Wno-initializer-overrides -O2 -pipe  -I/usr/local/include/python3.9 -fstack-protector-strong -fno-strict-aliasing   -fstack-protector-strong  -no-install -no-pie -o modules/python/tests/test_python_logmsg modules/python/tests/test_python_logmsg-test_python_logmsg.o ./libtest/libsyslog-ng-test.a -L/usr/local/lib -lcriterion  ./lib/libsyslog-ng.la -lm -lrt -L/usr/local/lib -Wl,--export-dynamic -lgmodule-2.0 -lglib-2.0 -lgthread-2.0 -pthread -lglib-2.0 -lintl  ./lib/eventlog/src/libevtlog.la ./lib/secret-storage/libsecret-storage.la   -L/usr/local/lib -lpcre  -lregex -ldl -dlpreopen ./modules/python/libmod-python.la -dlpreopen ./modules/syslogformat/libsyslogformat.la  
libtool:   error: only libraries may -dlpreopen a convenience library: './modules/python/libmod-python.la'
gmake[4]: *** [Makefile:15135: modules/python/tests/test_python_logmsg] Error 1
gmake[3]: *** [Makefile:26732: check-am] Error 2
gmake[2]: *** [Makefile:24985: check-recursive] Error 1
gmake[1]: *** [Makefile:26735: check] Error 2
```

## Steps to reproduce
'make check'

clang-14
FreeBSD 13.1

**Comment by bazsi** [comment]
This is strange as the rest of those test programs have used those linker options forever. There's only a slight ordering difference, can you check if this improves the situation?

```
diff --git a/modules/python/tests/Makefile.am b/modules/python/tests/Makefile.am
index e02e93e98..0c65d4986 100644
--- a/modules/python/tests/Makefile.am
+++ b/modules/python/tests/Makefile.am
@@ -16,8 +16,7 @@ modules_python_tests_TESTS = \
 modules_python_tests_test_python_logmsg_CFLAGS = $(TEST_CFLAGS) $(PYTHON_CFLAGS) -I$(top_srcdir)/modules/python
 modules_python_tests_test_python_logmsg_LDADD = $(TEST_LDADD) \
        -dlpreopen $(top_builddir)/modules/python/libmod-python.la \
-       $(PREOPEN_SYSLOGFORMAT) \
-       $(PYTHON_LIBS)
+       $(PYTHON_LIBS) $(PREOPEN_SYSLOGFORMAT)
 
 modules_python_tests_test_python_template_CFLAGS = $(TEST_CFLAGS) $(PYTHON_CFLAGS) \
        -I$(top_srcdir)/modules/python -I$(top_srcdir)/modules/syslogformat
```

**Comment by bazsi** [comment]
@yurivict can you check if the above patch helps, please?

**Comment by MrAnno** [comment]
This has been closed and hopefully resolved by #4232.
Please reopen if it's not the case.
